### PR TITLE
[WIP] Initial Azure Management system files for POC.  You will need to obta…

### PR DESCRIPTION
**Pull request by jteehan**
…in an Azure Subscription ID as well as the associated .pem file to execute the py.test test case examples.

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5c7c73d72b5d7b616855c5ad6b019acfe468846a:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/cloud/test_mgmt_sys_azure.py:1:1: F401 'pytest' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:3:1: F401 'flash' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:4:1: F401 'error' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:5:1: F401 'Credential' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:6:1: F401 'testgen' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:6:1: F401 'version' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:7:1: F401 'update' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:16:1: E302 expected 2 blank lines, found 1
./cfme/tests/cloud/test_mgmt_sys_azure.py:24:1: E302 expected 2 blank lines, found 1
./cfme/tests/cloud/test_mgmt_sys_azure.py:28:5: F841 local variable 'test_vm_state' is assigned to but never used
./cfme/tests/cloud/test_mgmt_sys_azure.py:30:1: E302 expected 2 blank lines, found 1
./cfme/tests/cloud/test_mgmt_sys_azure.py:34:5: F841 local variable 'test_vm_state' is assigned to but never used
./utils/virtual_machines.py:13:101: E501 line too long (109 > 100 characters)
./utils/mgmt_system/azuresms.py:6:1: F401 'winrm' imported but unused
./utils/mgmt_system/azuresms.py:18:1: F403 'from azure import *' used; unable to detect undefined names
./utils/mgmt_system/azuresms.py:19:1: F403 'from azure.servicemanagement import *' used; unable to detect undefined names
./utils/mgmt_system/azuresms.py:23:1: E303 too many blank lines (3)
./utils/mgmt_system/azuresms.py:49:9: F841 local variable 'subscription_id' is assigned to but never used
./utils/mgmt_system/azuresms.py:50:9: F841 local variable 'cert_file' is assigned to but never used
./utils/mgmt_system/azuresms.py:88:101: E501 line too long (144 > 100 characters)
./utils/mgmt_system/azuresms.py:89:101: E501 line too long (181 > 100 characters)
./utils/mgmt_system/azuresms.py:90:101: E501 line too long (133 > 100 characters)
./utils/mgmt_system/azuresms.py:105:101: E501 line too long (144 > 100 characters)
./utils/mgmt_system/azuresms.py:106:101: E501 line too long (181 > 100 characters)
./utils/mgmt_system/azuresms.py:107:101: E501 line too long (158 > 100 characters)
./utils/mgmt_system/azuresms.py:135:101: E501 line too long (144 > 100 characters)
./utils/mgmt_system/azuresms.py:136:101: E501 line too long (111 > 100 characters)
./utils/mgmt_system/azuresms.py:173:101: E501 line too long (126 > 100 characters)
```
:trollface:
*CFME QE Bot*

**Comment by mfalesni** [comment]
I see you might not have your dev environment set up properly since you got so many lint errors. Should you need any help, I can help.

**Comment by psav** [comment]
@jteehan http://cfme-tests.readthedocs.org/guides/lint.html May be of use here

**Comment by dajoRH** [comment]
Lint report for commit 1b6658184ffdda118a61e3174be998936411926f:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/cloud/test_mgmt_sys_azure.py:1:1: F401 'pytest' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:3:1: F401 'flash' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:4:1: F401 'error' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:5:1: F401 'Credential' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:6:1: F401 'testgen' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:6:1: F401 'version' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:7:1: F401 'update' imported but unused
./cfme/tests/cloud/test_mgmt_sys_azure.py:16:1: E302 expected 2 blank lines, found 1
./cfme/tests/cloud/test_mgmt_sys_azure.py:24:1: E302 expected 2 blank lines, found 1
./cfme/tests/cloud/test_mgmt_sys_azure.py:28:5: F841 local variable 'test_vm_state' is assigned to but never used
./cfme/tests/cloud/test_mgmt_sys_azure.py:30:1: E302 expected 2 blank lines, found 1
./cfme/tests/cloud/test_mgmt_sys_azure.py:34:5: F841 local variable 'test_vm_state' is assigned to but never used
./utils/virtual_machines.py:13:101: E501 line too long (109 > 100 characters)
./utils/mgmt_system/azuresms.py:6:1: F401 'winrm' imported but unused
./utils/mgmt_system/azuresms.py:18:1: F403 'from azure import *' used; unable to detect undefined names
./utils/mgmt_system/azuresms.py:19:1: F403 'from azure.servicemanagement import *' used; unable to detect undefined names
./utils/mgmt_system/azuresms.py:20:1: F401 'Credential' imported but unused
./utils/mgmt_system/azuresms.py:48:1: W293 blank line contains whitespace
./utils/mgmt_system/azuresms.py:90:101: E501 line too long (144 > 100 characters)
./utils/mgmt_system/azuresms.py:91:101: E501 line too long (181 > 100 characters)
./utils/mgmt_system/azuresms.py:92:101: E501 line too long (133 > 100 characters)
./utils/mgmt_system/azuresms.py:107:101: E501 line too long (144 > 100 characters)
./utils/mgmt_system/azuresms.py:108:101: E501 line too long (181 > 100 characters)
./utils/mgmt_system/azuresms.py:109:101: E501 line too long (158 > 100 characters)
./utils/mgmt_system/azuresms.py:137:101: E501 line too long (144 > 100 characters)
./utils/mgmt_system/azuresms.py:138:101: E501 line too long (111 > 100 characters)
./utils/mgmt_system/azuresms.py:175:101: E501 line too long (126 > 100 characters)
```
:trollface:
*CFME QE Bot*

File: cfme/tests/cloud/test_mgmt_sys_azure.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import pytest
+
+import cfme.web_ui.flash as flash
+import utils.error as error
+from cfme import Credential
+from utils import testgen, version
+from utils.update import update
+from utils.mgmt_system import azuresms
+
```
Comment (psav) [inline cfme/tests/cloud/test_mgmt_sys_azure.py]: In here, we'll need to use testgen to create the parametrization, basically we'll have something like this

```python
pytest_generate_tests = testgen.generate(testgen.provider_by_type, ['azure'],
                                         scope='module')
```

Then each of the tests would use
```python
def test_azure_vm_list(provider):
    provider.mgmt.list_vm('WS2012')
```
At least that's how it should work anyway :)

Comment (psav) [inline cfme/tests/cloud/test_mgmt_sys_azure.py]: This looks through all of the ```cfme_data.yaml``` and finds all providers with type "azure"

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by jteehan** [comment]
I'm going to withdraw this in leiu of a better solution that supports ARM mode.

**Pull request closed without merging**